synapse/federation/federation_client.py

```diff
@@ -1006,7 +1006,7 @@ async def _do_send_join(
                 content=pdu.get_pdu_json(time_now),
             )
         except HttpResponseException as e:
-            # If an error is received that is due to an unrecognised endpoint,
```

Comment:
I'm not sure this particular change is valid; the protocol might use american spelling for the error code but in a casual comment talking about unrecognised endpoints, I see no harm in writing them with native spelling (and indeed much of the code is using english spelling, so for consistency you might say that this would be more expected), much like we wouldn't force ourselves to misspell referrer if we were talking about the value of a 'Referer' header.